airootfs/usr/share/vx-img/flash-image.sh

```diff
@@ -78,31 +81,75 @@ for f in "$_path"/*; do
     fi
 done
 
+if [[ -z $_toflash ]]; then
+    echo "Found no image to flash. Exiting..."
+    exit
+fi
+
 echo "Extracting and flashing $_toflash"
 
-echo "What is the expected final size of the image? [64g]:"
+clear
+echo "What is the expected final size of the image, in GB? [64]:"
 read -r _finalsize
 
 if [[ -z $_finalsize ]]; then
-    _finalsize="64g"
+    _finalsize="64"
 fi
 
-$_compression -c -d $_path/"$_filename" | pv -s $_finalsize > /dev/nvme0n1
+_disk="/dev/nvme0n1"
+
+# Get our image size in raw bytes
+_size=$(numfmt --from=iec "${_finalsize}G")
+
+clear
+echo "Found the following disks to flash:"
+while true; do
+
+    # Get a list of all available disks large enough to take our image, sorted by size
+    IFS=$'\n' read -r -d '' -a disks <<< "$(lsblk -x SIZE -nblo NAME,SIZE,TYPE | grep "disk" | awk -v var="$_size" '$2 > var {print $1}')"
+
+    i=1
+    for disk in "${disks[@]}"; do
+        echo "$i. /dev/$disk"
+        ((i+=1))
+    done
+    
+
+    echo "Selecting /dev/${disks[-1]} to flash. Okay? [Y/n]:" 
```

Comment:
Lets make it 
`Which disk would you like to flash? Default: /dev/${disks[-1]}: ` 
Then if they just press enter we just item #1, so then if you want to do something else its not an extra step